troubadix/plugins/variable_redefinition_in_foreach.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+
+import re
+from pathlib import Path
+from typing import Iterator
+
+from troubadix.plugin import FileContentPlugin, LinterResult, LinterWarning
+
+FOREACH_PATTERN = re.compile(r"foreach\s+(?P<ident>\w+)\s*\((?P<iter>.+)\)")
+MAKE_LIST_PATTERN = re.compile(
+    r"^(?:make_list|make_list_unique)\((?P<params>.+)\)$"
+)
+
+
+class CheckVariableRedefinitionInForeach(FileContentPlugin):
+    name = "check_variable_redefinition_in_foreach"
+
+    def check_content(
+        self, nasl_file: Path, file_content: str
+    ) -> Iterator[LinterResult]:
+        """This plugin checks for a redefinition of the variable
+        that is passed to the foreach loop.
+        This can be caused by using same variable name
+        for both the list and the element being iterated over.
+        Incorrect uses of foreach loops that are covered by this plugin are:
+        foreach foo(foo){}
+        foreach foo(make_list(bar,foo)){}
+        foreach foo(make_list_unique(bar,foo)){}
+        """
+
+        for foreach_match in FOREACH_PATTERN.finditer(file_content):
+            identifier = foreach_match.group("ident")
+            iterator = foreach_match.group("iter").replace(" ", "")
```

Comment:
Replace instead of strip was done because of strings like 'make_list_unique( bar , foo )'. Then further calls to strip() would be necessary after the function parameters are split.
```suggestion
            iterator = foreach_match.group("iter").strip()
            
             if make_list_match := MAKE_LIST_PATTERN.fullmatch(iterator):
                make_list_params = make_list_match.group("params").split(",")
                stripped_params = [s.strip() for s in make_list_params]
                if identifier in stripped_params:
```

the list generator with strip() is new but not highlighted.

Reply:
added the comments in 0b1bc5ea1aeddcc13697c49bad590ea915828249